internal/pkg/otel/manager/manager.go

```diff
@@ -141,10 +141,12 @@ func (m *OTelManager) Run(ctx context.Context) error {
 				}
 				// ensure that the coordinator knows that there is no error
 				// as the collector is not running anymore
-				select {
-				case m.errCh <- nil:
-				case <-ctx.Done():
-				}
+				go func() {
```

Comment:
Why was this necessary? The write to the error channel blocking forever or taking a while is a hint that something else is wrong IMO, assuming that's why this changed.

Perhaps tied to my repeatable observation that the otel components take several seconds to show up in the output of `elastic-agent status`.

Reply:
@swiatekm Has identified this as a deadlock and changing this removes that deadlock. I am looking into why this is the case and if this can be removed/changed into a something different.

Reply:
I added a comment explaining this is only here to verify that integration tests pass with it present. I'll remove it once that's done, and we can fix the deadlock in a different PR.